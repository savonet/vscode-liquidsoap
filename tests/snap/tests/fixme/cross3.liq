#!/usr/bin/env -S dune exec ../../src/bin/liquidsoap.exe -- --no-stdlib -i

# See #2080. Simplified version. Here, "float" stands for "source".

# Our main source.
s = 3.

def once((s:float))
  s.{id = "bla"}
end

# Here, because of once, we infer that f should take as argument a source with
# many methods (id, etc.). However, we use it with cross below, which takes a
# function operating on undecorated sources.
def f(a)
  ignore([once(2.), a])
end

def cross((f : (float) -> unit), s)
  ignore(f)
  s
end

s = cross(f, s)

ignore(s)
