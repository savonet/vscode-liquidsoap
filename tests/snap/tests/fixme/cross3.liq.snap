>#!/usr/bin/env -S dune exec ../../src/bin/liquidsoap.exe -- --no-stdlib -i
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
>
># See #2080. Simplified version. Here, "float" stands for "source".
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
>
># Our main source.
#^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
>s = 3.
#^^^^ source.liquidsoap
#    ^ source.liquidsoap constant.numeric.integer
#     ^^ source.liquidsoap
>
>def once((s:float))
#^^^ source.liquidsoap keyword.control.liquidsoap
#   ^^^^^^^^^^^^^^^^^ source.liquidsoap
>  s.{id = "bla"}
#^^^^^^^^^^ source.liquidsoap
#          ^ source.liquidsoap string.quoted.double.liquidsoap
#           ^^^ source.liquidsoap string.quoted.double.liquidsoap
#              ^ source.liquidsoap string.quoted.double.liquidsoap
#               ^^ source.liquidsoap
>end
#^^^ source.liquidsoap keyword.control.liquidsoap
>
># Here, because of once, we infer that f should take as argument a source with
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
># many methods (id, etc.). However, we use it with cross below, which takes a
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
># function operating on undecorated sources.
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
>def f(a)
#^^^ source.liquidsoap keyword.control.liquidsoap
#   ^^^^^^ source.liquidsoap
>  ignore([once(2.), a])
#^^^^^^^^^^^^^^^ source.liquidsoap
#               ^ source.liquidsoap constant.numeric.integer
#                ^^^^^^^^ source.liquidsoap
>end
#^^^ source.liquidsoap keyword.control.liquidsoap
>
>def cross((f : (float) -> unit), s)
#^^^ source.liquidsoap keyword.control.liquidsoap
#   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
>  ignore(f)
#^^^^^^^^^^^^ source.liquidsoap
>  s
#^^^^ source.liquidsoap
>end
#^^^ source.liquidsoap keyword.control.liquidsoap
>
>s = cross(f, s)
#^^^^^^^^^^^^^^^^ source.liquidsoap
>
>ignore(s)
#^^^^^^^^^^ source.liquidsoap
>